test/test_SimulatedLinkamT95.py

```diff
@@ -0,0 +1,236 @@
+        self.assertFalse(ord(status_bytes[1]) & 0x01)
+
+        # Set up to cool from 24.0 C to 4.0 C at 51.00 C/min
+        linkam.setRate('5100')
+        linkam.setLimit('40')
+        linkam.start()
+        linkam.process(0)
+
+        # Ensure flag is set after running a bit
+        linkam.process(0.1)
+        status_bytes = linkam.getStatus()
+        self.assertTrue(ord(status_bytes[1]) & 0x01)
+
+    def test_stop_command(self):
+        linkam = SimulatedLinkamT95()
+        linkam.process(0)  # Initialize
+
+        # Issue T command to get into stopped state
+        linkam.getStatus()
+        linkam.process(0)
+
+        # Set up to heat from 24.0 C to 44.0 C at 20.00 C/min
+        linkam.setRate('2000')
+        linkam.setLimit('440')
+        linkam.start()
+        linkam.process(0)
+
+        # Process for some time and then stop
+        linkam.process(10)
+        linkam.stop()
+        linkam.process(0)
+
+        # Ensure status byte reports stopped
+        status_bytes = linkam.getStatus()
+        self.assertEqual(ord(status_bytes[0]),  0x01)
+
+    def test_hold_and_resume(self):
+        linkam = SimulatedLinkamT95()
+        linkam.process(0)  # Initialize
+
+        # Issue T command to get into stopped state
+        linkam.getStatus()
+        linkam.process(0)
+
+        # Set up to cool from 24.0 C to 4.0 C at 20.00 C/min
+        linkam.setRate('2000')
+        linkam.setLimit('40')
+        linkam.start()
+        linkam.process(0)
+
+        # Cool for a while
+        linkam.process(30)
+        status_bytes = linkam.getStatus()
+        self.assertEqual(status_bytes[0], '\x20')           # Cooling status set
+        self.assertNotEqual(status_bytes[6:10], '0028')     # Temp != 4.0 C
+
+        # Hold for a while
+        linkam.hold()
+        linkam.process(30)
+        status_bytes = linkam.getStatus()
+        self.assertEqual(status_bytes[0], '\x50')           # Manually holding
+        self.assertNotEqual(status_bytes[6:10], '0028')     # Temp != 4.0 C
+
+        # Cool some more
+        linkam.cool()
+        linkam.process(15)
+        status_bytes = linkam.getStatus()
+        self.assertNotEqual(status_bytes[6:10], '0028')  # Temp != 4.0 C
+
+        # Hold again
+        linkam.hold()
+        linkam.process(30)
+        status_bytes = linkam.getStatus()
+        self.assertEqual(status_bytes[0], '\x50')  # Manually holding
+        self.assertNotEqual(status_bytes[6:10], '0028')  # Temp != 4.0 C
+
+        # Finish cooling via heat command (should also work)
+        linkam.heat()
+        linkam.process(15)
+        status_bytes = linkam.getStatus()
+        self.assertEqual(status_bytes[6:10], '0028')  # Temp == 4.0 C
+
+        # Make sure transitions to auto-holding
+        linkam.process(0)
+        status_bytes = linkam.getStatus()
+        self.assertEqual(status_bytes[0], '\x30')  # Auto-holding at limit
+
+    def test_pump_command(self):
```

Comment:
Where would you add that?

Arguably, a function like that should be provided by `unittest.TestCase` (and it is a fairly common complaint that such a thing doesn't exist).

It doesn't really make sense to make a helper function like that in `TestSimulatedLinkamT95`, since chances are we'll want to use it in other tests as well.

A free-standing function in `core.utils` maybe? We'll still have to pass in the TestCase instance in order to call `TestCase.fail` when an exception is caught, right? Should we also pass in an expected exception type, the same way `TestCase.assertRaises` does?